lib/ansible/modules/network/ios/ios_ping.py

```diff
@@ -0,0 +1,212 @@
+    dest: 10.10.10.10
+
+- name: Test reachability to 10.20.20.20 using prod vrf
+  ios_ping:
+    provider: "{{ provider }}"
+    dest: 10.20.20.20
+    vrf: prod
+
+- name: Test unreachability to 10.30.30.30 using default vrf
+  ios_ping:
+    provider: "{{ provider }}"
+    dest: 10.30.30.30
+    state: absent
+
+- name: Test reachability to 10.40.40.40 using prod vrf and setting count and source
+  ios_ping:
+    provider: "{{ provider }}"
+    dest: 10.40.40.40
+    source: loopback0
+    vrf: prod
+    count: 20
+'''
+
+RETURN = '''
+commands:
+  description: Show the command sent.
+  returned: always
+  type: list
+  sample: ["ping vrf prod 10.40.40.40 count 20 source loopback0"]
+packet_loss:
+  description: Percentage of packets lost.
+  returned: always
+  type: str
+  sample: "0%"
+packets_rx:
+  description: Packets successfully received.
+  returned: always
+  type: int
+  sample: 20
+packets_tx:
+  description: Packets successfully transmitted.
+  returned: always
+  type: int
+  sample: 20
+rtt:
+  description: Show RTT stats.
+  returned: always
+  type: dict
+  sample: {"avg": 2, "max": 8, "min": 1}
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ios import run_commands
+from ansible.module_utils.ios import ios_argument_spec, check_args
+import re
+
+
+def main():
+    """ main entry point for module execution
+    """
+    argument_spec = dict(
+        count=dict(type="int"),
+        dest=dict(type="str", required=True),
+        source=dict(type="str"),
+        state=dict(type="str", choices=["absent", "present"], default="present"),
+        vrf=dict(type="str")
+    )
+
+    argument_spec.update(ios_argument_spec)
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    count = module.params["count"]
+    dest = module.params["dest"]
+    source = module.params["source"]
+    vrf = module.params["vrf"]
+
+    warnings = list()
+    check_args(module, warnings)
+
+    results = {}
+    if warnings:
+        results["warnings"] = warnings
+
+    results["commands"] = [build_ping(dest, count, source, vrf)]
+
+    ping_results = run_commands(module, commands=results["commands"])
+    ping_results_list = ping_results[0].split("\n")
+
+    success, rx, tx, rtt = parse_ping(ping_results_list[3])
+    loss = abs(100 - int(success))
+    results["packet_loss"] = str(loss) + "%"
+    results["packets_rx"] = int(rx)
+    results["packets_tx"] = int(tx)
+
+    # Convert rtt values to int
+    for k, v in rtt.items():
+        if rtt[k] is not None:
+            rtt[k] = int(v)
+
+    results["rtt"] = rtt
+
+    validate_results(module, loss, results)
+
+    module.exit_json(**results)
+
+
+def build_ping(dest, count=None, source=None, vrf=None):
+    """
+    Function to build the command to send to the terminal for the switch
+    to execute. All args come from the module's unique params.
+    """
+    if vrf is not None:
+        cmd = "ping {} {}".format(vrf, dest)
```

Comment:
For python 2.6 compatibility, this and the following four strings have to look like `cmd = "ping {0} {1}".format(vrf, dest)`